learn/tutorials/adding-typescript-views-to-a-vaadin-14-app/content.adoc

```diff
@@ -0,0 +1,429 @@
+
+.pom.xml excerpt with the Vaadin dependency declaration
+[source, xml]
+----
+<properties>
+
+    ...
+    <!-- Find the latest version on https://github.com/vaadin/platform/releases -->
+    <vaadin.version>15.0.0.alpha7</vaadin.version>
+    ...
+
+</properties>
+
+...
+
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-bom</artifactId>
+            <version>${vaadin.version}</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+----
+
+Run the Maven `install` goal to verify the changes above are correct, e. g., using the terminal command:
+
+[source, bash]
+----
+$ mvn install
+----
+
+== Add new frontend dependencies
+
+We will use link:https://vaadin.com/router[Vaadin Router] for client-side views routing, and also link:https://lit-element.polymer-project.org[lit-element] for creating TypeScript views.
+
+Run the following command to install these and add them to `package.json`:
+
+[source, bash]
+----
+$ npm install --save @vaadin/router lit-element
+----
+
+== Create client-side entrypoints
+
+Create the `frontend/index.html` and `frontend/index.ts` files.
+
+`index.html` is an HTML page template with metadata and initial content containing the client-side router outlet.
+
+.frontend/index.html
+[source,html]
+----
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <!-- Polyfills for the Web Component APIs are required to use Web Components
+     in the browsers that do not support these APIs natively (e.g. IE11) -->
+  <script src="./VAADIN/build/webcomponentsjs/webcomponents-loader.js" defer></script>
+  <style>
+    body, #outlet {
+      height: 100vh;
+      width: 100vw;
+      margin: 0;
+    }
+  </style>
+  <!-- index.ts is included here automatically (either by the dev server or during the build) -->
+</head>
+<body>
+  <div id="outlet"></djv>
+</body>
+</html>
+----
+
+The `index.ts` file is a main TypeScript frontend entrypoint. It uses `Vaadin Router` to enable client-side routing.
+
+.frontend/index.ts
+[source, typescript]
+----
+import {Flow} from '@vaadin/flow-frontend/Flow';
+import {Router} from '@vaadin/router';
+
+const {serverSideRoutes} = new Flow({
+  imports: () => import('../target/frontend/generated-flow-imports')
+});
+
+const routes = [
+  // fallback to server-side Flow routes if no client-side routes match
+  ...serverSideRoutes
+];
+
+const router = new Router(document.querySelector('#outlet'));
+router.setRoutes(routes);
+----
+
+Start your application in a regular way and open it in a browser to verify the changes are complete. Ideally, your application works as it did before the first step.
+
+== Convert your main layout to TypeScript
```

Comment:
This sounds like it will describe how to convert my main layout to TS. Instead it provides an example main layout written in TS. My main layout contains a lot of annotations for style loading, authentication handling, menu generation and a lot of other stuff. I would be disappointed

Reply:
It’s hard here to cover these topics in detail, as these are app-specific things. So having a simple layout example to follow seems more reasonable.

The menu generation and styling the layout itself are covered, moving the annotations is covered in the example though.